Create 2019-12-03-countdown-to-the-new-year-ember-modifier.md (#399)
ember-learn/ember-blog: source/2019-12-02-countdown-to-the-new-year-ember-modifier.md

> ---

Reviewer: I think to reflect the correct date on the blog, this file would need to be named `source/2019-12-03-countdown-to-the-new-year-ember-modifier.md` - what do you think?

Author: oh good catch! silly mistake!

title: Countdown to The New Year- Ember Modifier
author: Melanie Sumner

> tags: 2019, Addons, DecEmber

Reviewer: I think we can also add an alias here:

```suggestion
tags: 2019, Addons, DecEmber
alias: "blog/2019/12/03/countdown-to-the-new-year-2019-ember-modifier"
```

responsive: true
---

## Day 3

This is the third in our DecEmber series– _"Countdown to The New Year: 31 Days of Ember Addons"_. We plan to highlight a new addon each day until the new year, and we hope you'll join us for the fun!

Today we are going to look at a rising star in our addon ecosystem- [ember-modifier](https://emberobserver.com/addons/ember-modifier). 

READMORE

### What It Does

> The `ember-modifier` addon provides an API for authoring element modifiers in Ember. It mirrors Ember's [helper API](https://api.emberjs.com/ember/3.14/classes/Ember.Templates.helpers), and has forms for writing both simple functional modifiers and more complicated class modifiers.

Reviewer 2: It could be nice to add an example here. Unless someone is using Octane, they don't know what a modifier is or what kinds of problems it solves.

Reviewer 2: A modifier can add special, automatic behavior to a DOM element, such as tracking scroll position or adding new event listeners. After you write a modifier, you can use it in a template like this:

```
<div {{my-modifier}}>
  Some more content
</div>
```

 Developers who have tried out the Octane preview might already be familiar with [ember-render-modifiers](https://github.com/emberjs/ember-render-modifiers), which provide some modifiers like `{{did-insert}}` out of the box. With ember-modifiers, you can go beyond the box to build whatever your app needs!

Author: Great idea, thank you @jenweber!

> For an in-depth look at modifiers in Ember, checkout [pzuraq's](https://github.com/pzuraq) blog post about [modifiers in Ember Octane](https://blog.emberjs.com/2019/03/06/coming-soon-in-ember-octane-part-4.html). 

Reviewer 3: ```suggestion
For an in-depth look at modifiers in Ember, check out [pzuraq's](https://github.com/pzuraq) blog post about [modifiers in Ember Octane](https://blog.emberjs.com/2019/03/06/coming-soon-in-ember-octane-part-4.html). 
```

---
title: Countdown to The New Year- Ember Modifier
author: Melanie Sumner
tags: 2019, Addons, DecEmber
responsive: true
---

## Day 3

This is the third in our DecEmber series– _"Countdown to The New Year: 31 Days of Ember Addons"_. We plan to highlight a new addon each day until the new year, and we hope you'll join us for the fun!

Today we are going to look at a rising star in our addon ecosystem- [ember-modifier](https://emberobserver.com/addons/ember-modifier). 

READMORE

### What It Does

The `ember-modifier` addon provides an API for authoring element modifiers in Ember. It mirrors Ember's [helper API](https://api.emberjs.com/ember/3.14/classes/Ember.Templates.helpers), and has forms for writing both simple functional modifiers and more complicated class modifiers.

> A modifier can add special, automatic behavior to a DOM element, such as tracking scroll position or adding new event listeners. After you write a modifier, you can use it in a template like this:

Reviewer: I think the linter error on CI refers to one of the words in the post right now:

```
source/2019-12-02-countdown-to-the-new-year-ember-modifier.md
  20:20-20:27  warning  `special` may be insensitive, use `has a disability`, `person with a disability`, `people with disabilities` instead  special  retext-equality
```

This suggestion should fix it I think:

```suggestion
<!--alex ignore special-->
A modifier can add special, automatic behavior to a DOM element, such as tracking scroll position or adding new event listeners. After you write a modifier, you can use it in a template like this:
```

Author: Changed the world to "customized" - what do you think?

Reviewer: Yes, that sounds awesome!
